flink-streaming-java/src/main/java/org/apache/flink/streaming/api/operators/StreamOperatorStateHandler.java

```diff
@@ -269,9 +273,14 @@ void snapshotState(
         }
     }
 
+    private boolean isCanonicalSavepoint(SnapshotType snapshotType) {
```

Comment:
I just share the concern that using an explicit cast to SavepointType here doesn't look so good. But I see that it is problem with current behavior in general, not with this PR.

Reply:
Yes, I also did not like the casting, therefore I am more than happy to hear alternatives.